File: Calculadora_Financeira.java
import java.util.Scanner;

public class Calculadora_Financeira {

    private static double calcularValorFuturo(double pv, double i, int n) {
        return pv * Math.pow(1 + i, n);
    }

    private static double calcularValorPresente(double vf, double i, int n) {
        return vf / Math.pow(1 + i, n);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Imprimir valores para o calculo de Valor Futuro e Valor Presente:");
        System.out.println("______________________________________________________________________");

        // Código referente ao cálculo do Valor presente
        System.out.print("\nDigite o valor futuro que almeja: R$:");
        double vfDesejado = scanner.nextDouble();
        System.out.print("\nDigite a taxa de juros anual (em %):");
        double i = scanner.nextDouble() / 100;
        System.out.print("\nDigite a quantidade de anos que deseja calcular:");
        int n = scanner.nextInt();
        double pvDesejado = calcularValorPresente(vfDesejado, i, n);
        System.out.printf("\nPara obter R$%.2f apos %d anos, voce precisa depositar: R$%.2f\n",
                vfDesejado, n, pvDesejado);
        System.out.println("______________________________________________________________________");
        
        // Código referente ao cálculo do Valor Futuro
        System.out.print("\nDigite o valor presente referente ao deposito inicial: R$:");
        double pv = scanner.nextDouble();
        System.out.print("\nDigite a taxa de juros anual(em %):");
        i = scanner.nextDouble() / 100;
        System.out.print("\nDigite a quantidade de anos que deseja calcular:");
        n = scanner.nextInt();
        double vf = calcularValorFuturo(pv, i, n);
        System.out.printf("\nO valor futuro apos %d anos sera: R$%.2f\n", n, vf);
        System.out.println("______________________________________________________________________");
        
        System.out.printf("\n");
        scanner.close();
    }
}

File: calculadora_financeira.py
def calcular__valor__futuro(pv, i, n):
    vf = pv * (1 + i) ** n
    return vf

def calcular__valor__presente(vf, i, n):
    pv = vf / (1 + i) ** n
    return pv

print("Cálculo de Valor Futuro e Valor Presente")
print("__________________________________________________________________________________________________")


vf_desejado = float(input("\nDigite o valor futuro que quer R$:"))
i = float(input("\nDigite a taxa de juros anual em %:")) / 100
n = int(input("\nDigite a quantidade de anos para calcular:"))


pv_desejado = calcular__valor__presente(vf_desejado, i, n)
print(f"\nPara obter ${vf_desejado:.2f} após {n} anos, você precisaria depositar: ${pv_desejado:.2f}\n")
print("__________________________________________________________________________________________________")


pv = float(input("\nDigite o valor presente referente ao deposito inicial R$:"))
i = float(input("\nDigite a taxa de juros anual em %:")) / 100
n = int(input("\nDigite a quantidade de anos para calcular:"))


vf = calcular__valor__futuro(pv, i, n)
print(f"\nO valor futuro apos {n} anos sera: ${vf:.2f}\n")
print("__________________________________________________________________________________________________")

